src/Domain/Block.cpp

```diff
@@ -34,8 +35,17 @@ Block<VolumeDim>::Block(
   // Loop over Directions to search which Directions were not set to neighbors_,
   // set these Directions to external_boundaries_.
   for (const auto& direction : Direction<VolumeDim>::all_directions()) {
+    const bool has_boundary_in_this_direction = has_boundary(
+        gsl::at(topologies_, direction.dimension()), direction.side());
     if (neighbors_.find(direction) == neighbors_.end()) {
-      external_boundaries_.emplace(direction);
+      if (has_boundary_in_this_direction) {
+        external_boundaries_.emplace(direction);
+      }
```

Comment:
You can also assert here that the number of neighbors is 1 for cubes and either 1 or 6 for shells

Reply:
no, that is not true in general